### PR TITLE
HC-482: Make User Rule Job evaluation more targeted

### PR DESCRIPTION
**Pull request by DustinKLo**
Change(s):
- passing the job doc's index to userrules task
- removed alias 'job_status' in favor of `job_status-current` in user rules job evaluation

Related ticket:
- https://hysds-core.atlassian.net/browse/HC-482

Related PR(s):
- https://github.com/hysds/lightweight-jobs/pull/29

Jenkins:
- dev-e2e (reproc): https://nisar-pcm-ci.jpl.nasa.gov/job/force-branches/job/ci-force_branch-E2E-test/272/
- dev-e2e (fwd): https://nisar-pcm-ci.jpl.nasa.gov/job/force-branches/job/ci-force_branch-E2E-test/273/

`job_worker.py` will now pass the index to the user rules job evaluation, the logs:
```
[2023-07-12 23:15:39,472: INFO/ForkPoolWorker-1] hysds.task_worker.run_task[f31be517-7e03-49ac-8aef-7c02229db2f8]: Final query: {"query": {"bool": {"must": [{"bool": {"must": [{"bool": {"must": [{"term": {"job.type": "job-lightweight-echo:v1.2.4"}}]}}]}}, {"term": {"_id": "fe5030be-0bf9-4020-85bd-3b497afb4b4a"}}]}}}
[2023-07-12 23:15:39,472: INFO/ForkPoolWorker-1] hysds.task_worker.run_task[f31be517-7e03-49ac-8aef-7c02229db2f8]: updated query: "{\"query\": {\"bool\": {\"must\": [{\"bool\": {\"must\": [{\"bool\": {\"must\": [{\"term\": {\"job.type\": \"job-lightweight-echo:v1.2.4\"}}]}}]}}, {\"term\": {\"_id\": \"fe5030be-0bf9-4020-85bd-3b497afb4b4a\"}}]}}}"
[2023-07-12 23:15:39,475: INFO/ForkPoolWorker-1] POST http://100.104.10.203:9200/job_status-2023.07.12/_search [status:200 request:0.003s]
[2023-07-12 23:15:39,475: INFO/ForkPoolWorker-1] hysds.task_worker.run_task[f31be517-7e03-49ac-8aef-7c02229db2f8]: Rule 'dustin_test_rule' successfully matched for fe5030be-0bf9-4020-85bd-3b497afb4b4a
[2023-07-12 23:15:39,502: INFO/ForkPoolWorker-1] hysds.task_worker.run_task[f31be517-7e03-49ac-8aef-7c02229db2f8]: Trigger task submitted for fe5030be-0bf9-4020-85bd-3b497afb4b4a: hysds-io-lw-mozart-notify-by-email:v1.2.4
```

if you look at where ES is called, instead of using the alias `job_status` it now uses the specific index (`job_status-2023.07.12`) where the document resides:
```
[2023-07-12 23:15:39,475: INFO/ForkPoolWorker-1] POST http://100.104.10.203:9200/job_status-2023.07.12/_search [status:200 request:0.003s]
```

### CODE REVIEW

**Review by pymonger — APPROVED** [review]
LGTM. Let's get @mcayanan to take a look as well.

File: hysds/user_rules_job.py [hunk 1]
```diff
@@ -19,12 +19,12 @@
 from elasticsearch import ElasticsearchException
 
 JOBS_ES_URL = app.conf.JOBS_ES_URL  # ES
-STATUS_ALIAS = app.conf.STATUS_ALIAS
```
Comment (pymonger) [inline hysds/user_rules_job.py:22]: @DustinKLo: did a quick sanity check on the other core repos and found this:
```
$ cd lightweight-jobs
$ grep -r STATUS_ALIAS .
./purge.py:        es_index = app.conf["STATUS_ALIAS"]
./retry.py:STATUS_ALIAS = app.conf["STATUS_ALIAS"]
./retry.py:    return mozart_es.search(index=STATUS_ALIAS, body=query_json)
./lib/get_component_configuration.py:        query_idx = app.conf["STATUS_ALIAS"]
./notify_by_email.py:        index = app.conf["STATUS_ALIAS"]
```
Should there also be a PR on lightweight-jobs?

Comment (DustinKLo) [reply hysds/user_rules_job.py:22]: @pymonger i think `retry.py` and `purge.py` would need to be updated
`get_component_configuration.py` isn't used anywhere and `notify_by_email.py`

File: hysds/user_rules_job.py [hunk 2]
```diff
@@ -79,14 +79,14 @@ def update_query(job_id, rule):
     return final_query
 
 
-def evaluate_user_rules_job(job_id, alias=STATUS_ALIAS):
+def evaluate_user_rules_job(job_id, index=None):
```
Comment (mcayanan) [inline hysds/user_rules_job.py:82]: Should we default to JOB_STATUS_ALIAS?

```suggestion
def evaluate_user_rules_job(job_id, index=JOB_STATUS_ALIAS):
```

File: hysds/user_rules_job.py [hunk 3]
```diff
@@ -141,12 +141,13 @@ def evaluate_user_rules_job(job_id, alias=STATUS_ALIAS):
 @backoff.on_exception(
     backoff.expo, socket.error, max_tries=backoff_max_tries, max_value=backoff_max_value
 )
-def queue_finished_job(_id):
+def queue_finished_job(_id, index=None):
```
Comment (mcayanan) [inline hysds/user_rules_job.py:144]: Should we default to JOB_STATUS_ALIAS?

```suggestion
def evaluate_user_rules_job(job_id, index=JOB_STATUS_ALIAS):
```

File: hysds/user_rules_job.py (hunk 2 above)
Comment (DustinKLo) [reply hysds/user_rules_job.py:82]: we default to `JOB_STATUS_ALIAS` when evaluating the query here:
```python
result = mozart_es.es.search(index=index or JOB_STATUS_ALIAS, body=final_qs)
```

**Review by pymonger — APPROVED** [review]

File: hysds/user_rules_job.py (hunk 2 above)
Comment (pymonger) [reply hysds/user_rules_job.py:82]: @mcayanan: does that work for you?

File: hysds/user_rules_job.py (hunk 3 above)
Comment (pymonger) [reply hysds/user_rules_job.py:144]: @mcayanan: should the suggestion be:
```suggestion
def queue_finished_job(_id, index=JOB_STATUS_ALIAS):
```

File: hysds/user_rules_job.py (hunk 2 above)
Comment (mcayanan) [reply hysds/user_rules_job.py:82]: oh my bad. I didn't see that. No change needed then. I'll resolve.

File: hysds/user_rules_job.py (hunk 3 above)
Comment (mcayanan) [reply hysds/user_rules_job.py:144]: yes. sorry I had a typo in my suggestion.

**Pull request merged**